conftest.py

```diff
@@ -76,46 +80,81 @@ def sample_dag():
+
+
 @pytest.fixture
 def test_table(request, sql_server):
-    is_tmp_table = True
-    table_options = {}
+    tables = []
+    tables_params = [{"is_temp": True}]
 
     if getattr(request, "param", None):
-        is_tmp_table = request.param.get("is_temp", True)
-        table_options = request.param.get("param", {})
+        if isinstance(request.param, list):
+            tables_params = request.param
+        else:
+            tables_params = [request.param]
+
     sql_name, hook = sql_server
     database = get_database_name(hook)
 
-    if database == Database.SNOWFLAKE:
-        params = {
-            "conn_id": hook.snowflake_conn_id,
-            "database": hook.database,
-            "warehouse": hook.warehouse,
-            "schema": hook.schema,
-        }
-    elif database == Database.POSTGRES:
-        params = {"conn_id": hook.postgres_conn_id, "database": hook.schema}
-    elif database == Database.SQLITE:
-        params = {"conn_id": hook.sqlite_conn_id, "database": "sqlite"}
-    elif database == Database.BIGQUERY:
-        params = {"conn_id": hook.gcp_conn_id, "schema": SCHEMA}
-    else:
-        raise ValueError("Unsupported Database")
-    params.update(table_options)
-    table = TempTable(**params) if is_tmp_table else Table(**params)
-
-    yield table
-
-    if table.qualified_name():
-        hook.run(f"DROP TABLE IF EXISTS {table.qualified_name()}")
-
-    if database == Database.SQLITE:
-        hook.run(f"DROP INDEX IF EXISTS unique_index")
-    elif database in (Database.POSTGRES, Database.POSTGRESQL):
-        # There are some tests (e.g. test_agnostic_merge.py) which create stuff which are not being deleted
-        # Example: tables which are not fixtures and constraints. This is an aggressive approach towards tearing down:
-        hook.run(f"DROP SCHEMA IF EXISTS {table.schema} CASCADE;")
+    for table_param in tables_params:
+        is_tmp_table = table_param.get("is_temp", True)
+        load_table = table_param.get("load_table", False)
+        override_table_options = table_param.get("param", {})
+
+        if database == Database.SNOWFLAKE:
+            default_table_options = {
+                "conn_id": hook.snowflake_conn_id,
+                "database": hook.database,
+                "warehouse": hook.warehouse,
+                "schema": hook.schema,
+            }
+        elif database == Database.POSTGRES:
+            default_table_options = {
+                "conn_id": hook.postgres_conn_id,
+                "database": hook.schema,
+            }
+        elif database == Database.SQLITE:
+            default_table_options = {
+                "conn_id": hook.sqlite_conn_id,
+                "database": "sqlite",
+            }
+        elif database == Database.BIGQUERY:
+            default_table_options = {"conn_id": hook.gcp_conn_id, "schema": SCHEMA}
+        else:
+            raise ValueError("Unsupported Database")
```

Comment:
Since we're not covering this line, and this is part of the test suite, we may want to delete it - what are your thoughts?

Reply:
That's fine, @utkarsharma2 , it's a one-liner - in the tests :)